### PR TITLE
Add named meter and meter factory

### PR DESCRIPTION
**Pull request by ptravers**
this resolves https://github.com/open-telemetry/opentelemetry-java/issues/588

### CODE REVIEW

**Comment by ptravers** [comment]
I signed it

**Comment by ptravers** [comment]
Thank you I will repair that this evening. Would you like me to wait till the updates to the metrics API are complete?

**Comment by arminru** [comment]
> Thank you I will repair that this evening. Would you like me to wait till the updates to the metrics API are complete?

Thanks! It would be fine with me if you merge the status quo (once reviewed by the others) to spare you the merge conflicts you'd encounter by leaving your PR open until the metrics SDK spec is finalized. With #670 open we won't forget about finishing the implementation.

File: api/src/main/java/io/opentelemetry/metrics/CounterDouble.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
  * <pre>{@code
  * class YourClass {
  *
- *   private static final Meter meter = OpenTelemetry.getMeter();
+ *   private static final Meter meter = OpenTelemetry.getMeterFactory().get("some_counter_double");
```
Comment (bogdandrutu) [inline api/src/main/java/io/opentelemetry/metrics/CounterDouble.java]: use "my_library_name" as example, and everywhere.

File: api/src/main/java/io/opentelemetry/metrics/GaugeDouble.java [hunk 2]
```diff
@@ -29,7 +29,7 @@
  * <pre>{@code
  * class YourClass {
  *
- *   private static final Meter meter = OpenTelemetry.getMeter();
+ *   private static final Meter meter = OpenTelemetry.getMeterFactory().get("some_gauge_double");
```
Comment (bogdandrutu) [inline api/src/main/java/io/opentelemetry/metrics/GaugeDouble.java]: ditto

File: api/src/main/java/io/opentelemetry/metrics/ObserverDouble.java [hunk 3]
```diff
@@ -26,7 +26,7 @@
  * <pre>{@code
  * class YourClass {
  *
- *   private static final Meter meter = OpenTelemetry.getMeter();
+ *   private static final Meter meter = OpenTelemetry.getMeterFactory().get("some_observer_double");
```
Comment (bogdandrutu) [inline api/src/main/java/io/opentelemetry/metrics/ObserverDouble.java]: ditto

File: exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/Adapter.java [hunk 4]
```diff
@@ -174,15 +175,19 @@ private Adapter() {}
     switch (value.getType()) {
       case STRING:
         builder.setVStr(value.getStringValue());
+        builder.setVType(Model.ValueType.STRING);
```
Comment (bogdandrutu) [inline exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/Adapter.java]: This was done in a separate PR, can you rebase to remove these changes?

File: exporters/jaeger/src/test/java/io/opentelemetry/exporters/jaeger/AdapterTest.java [hunk 5]
```diff
@@ -85,19 +85,20 @@ public void testProtoSpan() {
     assertEquals(duration, Durations.toMillis(jaegerSpan.getDuration()));
 
     assertEquals(4, jaegerSpan.getTagsCount());
-    Model.KeyValue keyValue = getValue(jaegerSpan.getTagsList(), "span.kind");
+    Model.KeyValue keyValue = getValue(jaegerSpan.getTagsList(), Adapter.KEY_SPAN_KIND);
```
Comment (bogdandrutu) [inline exporters/jaeger/src/test/java/io/opentelemetry/exporters/jaeger/AdapterTest.java]: same

File: api/src/main/java/io/opentelemetry/metrics/CounterDouble.java (hunk 1 above)
Comment (ptravers) [reply api/src/main/java/io/opentelemetry/metrics/CounterDouble.java]: sure thing!

File: exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/Adapter.java (hunk 4 above)
Comment (ptravers) [reply exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/Adapter.java]: yeah I tried to rebase early this morning and flubbed something will fix this now

File: api/src/main/java/io/opentelemetry/metrics/CounterDouble.java (hunk 1 above)
Comment (arminru) [reply api/src/main/java/io/opentelemetry/metrics/CounterDouble.java]: Maybe something like `com.example.MyLibraryName` would be better to indicate that a Java package name is expected here.

Comment (ptravers) [reply api/src/main/java/io/opentelemetry/metrics/CounterDouble.java]: yeah that seems very helpful!
